Core/src/org/sleuthkit/autopsy/modules/plaso/PlasoIngestModule.java

```diff
@@ -0,0 +1,350 @@
+import org.sleuthkit.datamodel.SleuthkitCase;
+import org.sleuthkit.datamodel.TimeUtilities;
+
+/**
+ * Data source ingest module that runs plaso against the image 
+ */
+
+public class PlasoIngestModule implements DataSourceIngestModule {
+
+    private static final Logger logger = Logger.getLogger(PlasoIngestModule.class.getName());
+    private static final String PLASO = "plaso";
+    private static final String PLASO64 = "plaso//plaso-20180127-amd64";
+    private static final String PLASO32 = "plaso//plaso-20180127-win32";
+    private static final String LOG2TIMELINE_EXECUTABLE = "Log2timeline.exe"; 
+    private static final String PSORT_EXECUTABLE = "psort.exe";
+    private static final String VSS_OPTIONS = "--vss-stores";
+    private static final String PARTITIONS = "--partitions";
+    private static final String HASHER_FILE_SIZE_LIMIT = "--hasher_file_size_limit";
+    private static final String ONE = "1";
+    private static final String HASHERS = "--hashers";
+    private static final String NONE = "none";
+    private static final String ALL = "all";
+    private static final String TIME_ZONE_PARM = "-z";
+    private IngestJobContext context;
+    private final Case currentCase = Case.getCurrentCase();
+    private final FileManager fileManager = currentCase.getServices().getFileManager();
+
+    PlasoIngestModule() {
+    }
+
+    @Override
+    public void startUp(IngestJobContext context) throws IngestModuleException {
+        this.context = context;
+
+    }
+
+    @NbBundle.Messages({
+        "PlasoIngestModule_startUp_message=Starting Plaso Run.",
+        "PlasoIngestModule_error_running=Error running Plaso, see log file.",
+        "PlasoIngestModule_log2timeline_cancelled=Log2timeline run was canceled",
+        "PlasoIngestModule_psort_cancelled=psort run was canceled",
+        "PlasoIngestModule_bad_imageFile=Cannot find image file name and path",
+        "PlasoIngestModule_log2timeline_executable_not_found=Log2timeline Executable Not Found",
+        "PlasoIngestModule_psort_executable_not_found=psort Executable Not Found",
+        "PlasoIngestModule_dataSource_not_an_image=Datasource is not an Image.",
+        "PlasoIngestModule_running_log2timeline=Running Log2timeline",
+        "PlasoIngestModule_running_psort=Running Psort",
+        "PlasoIngestModule_completed=Plaso Processing Completed",
+        "PlasoIngestModule_has_run=Plaso Plugin has been run.",
+    })
+
+    @Override
+    public ProcessResult process(Content dataSource, DataSourceIngestModuleProgress statusHelper) {
+        statusHelper.switchToIndeterminate();
+        Image img;
+        if (dataSource instanceof Image) {
+           img = (Image) dataSource;
+        } else {
+            logger.log(Level.SEVERE, Bundle.PlasoIngestModule_dataSource_not_an_image());
+            MessageNotifyUtil.Message.info(Bundle.PlasoIngestModule_error_running());
+            return ProcessResult.ERROR;                   
```

Comment:
This should probably just be OK. ERROR is more for critical system errors, not files that we choose to ignore.

Reply:
Will do